embassy-nrf/src/usb.rs

```diff
@@ -595,29 +619,19 @@ impl<'d, T: Instance> driver::ControlPipe for ControlPipe<'d, T> {
             let req = self.request.unwrap();
             assert_eq!(req.direction, UsbDirection::Out);
             assert!(req.length > 0);
-            assert!(buf.len() >= usize::from(req.length));
-
-            let regs = T::regs();
 
-            // Wait until ready
-            regs.intenset.write(|w| w.ep0datadone().set());
-            poll_fn(|cx| {
-                EP_OUT_WAKERS[0].register(cx.waker());
-                let regs = T::regs();
-                if regs
-                    .events_ep0datadone
-                    .read()
-                    .events_ep0datadone()
-                    .bit_is_set()
-                {
-                    Poll::Ready(())
-                } else {
-                    Poll::Pending
+            let req_length = usize::from(req.length);
```

Comment:
I think this chunking could go in `embassy-usb` instead of in the driver. It's the same for all drivers and AFAIK no hardware does it for you. If we make it the driver's responsibility it'll end up copypasted in all the drivers.

Same applies to `accept_in` from a previous PR, so I'm OK with merging this here and moving later if you prefer.